Sources/GRPCHTTP2Core/GRPCStreamStateMachine.swift

```diff
@@ -1019,10 +1042,6 @@ extension GRPCStreamStateMachine {
       headers.add(outboundEncoding.name, forKey: .encoding)
     }
 
-    for acceptedEncoding in configuration.acceptedEncodings.elements.filter({ $0 != .none }) {
-      headers.add(acceptedEncoding.name, forKey: .acceptEncoding)
-    }
```

Comment:
I don't think this is the right fix: there are times when we _should_ include the accept encoding headers, usually when the client tries to use an encoding we don't support (i.e. client uses compression and server isn't configured for compression).

It looks like there's a test bug: in `makeServerStateMachine`, the compression is unconditionally used, even if `compressionEnabled: false`.

Reply:
I went back to the RFC and this header appears as optional in all responses: `Response-Headers → HTTP-Status [Message-Encoding] [Message-Accept-Encoding] Content-Type *Custom-Metadata`. I think I had decided to always include it to make the logic simpler, as the RFC does not specify when and when it shouldn't be present. It's okay if we want to save some bytes and not send it unless the client tries to use an encoding we don't support.

WRT a bug being present in the tests: I don't think there is - it's just that I was always adding the accept-encoding header to the server's initial metadata. This doesn't mean the server will use compression, it's just advertising to the client that it can understand it. If I make the change to only send this header when we get an unsupported algorithm from the client though, I'll make the change to not always send it back.

Reply:
Actually, we _are_ including the accept-encoding header in the case you mentioned - it just happens in a different codepath because we are returning trailers instead of response headers: see line 1314 to 1333 in the state machine. We also have a test for this, `testReceiveMetadataWhenClientIdleAndServerIdle_ServerUnsupportedEncoding`.
I have fixed the initial server metadata in the tests though to not include the accept-encoding header unless compression is enabled, and to not accept deflate encoding unless the compressionEnabled flag is passed - both for consistency.

Reply:
> I think I had decided to always include it to make the logic simpler, as the RFC does not specify when and when it shouldn't be present. It's okay if we want to save some bytes and not send it unless the client tries to use an encoding we don't support.

Yeah, that's fine with me.

> WRT a bug being present in the tests: I don't think there is - it's just that I was always adding the accept-encoding header to the server's initial metadata. This doesn't mean the server will use compression, it's just advertising to the client that it can understand it. 

The code I originally commented on setup a server state machine which _didn't_ using compression but still advertised compression in the accept encoding header it sent to the client. If it doesn't support compression then it mustn't advertise to the client that it does.